File: backend/bookingmovieticketapp/app/Models/Payment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Payment extends Model
{
    use HasFactory;
    protected $table = 'payments';
    protected $primaryKey = 'id';
    public $incrementing = true;
    public $timestamps = false;
    protected $fillable = [
        'bookingid',
        'totalprice',
        'paymentmethod',
        'paymentstatus',
        'paymenttime',
    ];

    public function booking()
    {
        return $this->belongsTo(Booking::class, 'bookingid');
    }
}

File: frontendapp/lib/models/payment.dart
class Payment {
  final int? id;
  final int bookingId;
  final double totalPrice;
  final String paymentMethod;
  final String paymentStatus;
  final String paymentTime;

  Payment({
    this.id,
    required this.bookingId,
    required this.totalPrice,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.paymentTime,
  });

  factory Payment.fromMap(Map<String, dynamic> map) => Payment(
    id: map['ID'],
    bookingId: map['BOOKINGID'],
    totalPrice: map['TOTALPRICE'].toDouble(),
    paymentMethod: map['PAYMENTMETHOD'],
    paymentStatus: map['PAYMENTSTATUS'],
    paymentTime: map['PAYMENTTIME'],
  );

  Map<String, dynamic> toMap() => {
    'ID': id,
    'BOOKINGID': bookingId,
    'TOTALPRICE': totalPrice,
    'PAYMENTMETHOD': paymentMethod,
    'PAYMENTSTATUS': paymentStatus,
    'PAYMENTTIME': paymentTime,
  };
}
